packages/asset-swapper/src/swap_quoter.ts

```diff
@@ -165,14 +167,19 @@ export class SwapQuoter {
         this.orderbook = orderbook;
         this.expiryBufferMs = expiryBufferMs;
         this.permittedOrderFeeTypes = permittedOrderFeeTypes;
-        this._rfqtTakerApiKeyWhitelist = options.rfqt ? options.rfqt.takerApiKeyWhitelist || [] : [];
+        this._rfqtTakerApiKeyWhitelist = rfqt ? rfqt.takerApiKeyWhitelist || [] : [];
+        this._rfqtSkipBuyRequests =
+            // tslint:disable-next-line:no-non-null-assertion these are clearly defined in the declaration
+            (rfqt && rfqt.skipBuyRequests) || constants.DEFAULT_SWAP_QUOTER_OPTS.rfqt!.skipBuyRequests!;
```

Comment:
This feels a little weird: if `constants.DEFAULT_SWAP_QUOTER_OPTS.rfqt.skipBuyRequests` was true, but `rfqt` was explicitly defined with `{ skipBuyRequests: false }`, this line would still evaluate to `true`.

Could this instead be `rfqt ? rfqt.skipBuyRequests : constants.DEFAULT_SWAP_QUOTER_OPTS.rfqt!.skipBuyRequests!`

Reply:
Also -- I suggest removing `: SwapQuoterOpts` in the `const DEFAULT_SWAP_QUOTER_OPTS: SwapQuoterOpts = {` line in `constants.ts`

This would allow `constants.DEFAULT_SWAP_QUOTER_OPTS.rfqt!.skipBuyRequests!` to be changed to `constants.DEFAULT_SWAP_QUOTER_OPTS.rfqt.skipBuyRequests`

Reply:
Great catch, thank you.  Unfortunately `skipBuyRequests` is itself optional, so I couldn't use your suggestion verbatim, but it's pretty close: https://github.com/0xProject/0x-monorepo/pull/2574/commits/a9e72085dd5c62409e0433f44a2d9da21c386d70

As for removing that type annotation in `constants.ts`...  In my mind this is a `tslint` bug: it should be looking at the actual declaration of that `constants.DEFAULT...` and seeing that these things actually are defined.  Removing a type annotation just to satisfy a linter doesn't seem right to me.

Reply:
The `skipBuyRequests` looks better 👍 

With regards to `constants.ts`: this is not a tslint bug.  This is TypeScript obeying what was defined.  `constants.DEFAULT_SWAP_QUOTER_OPTS` is explicitly defined by us as type `SwapQuoterOpts `, and `SwapQuoterOpts` considers `rfqt` optional.  Thus -- when you reference this var, it doesn't know if `rfqt` is defined or not, because it's relying on the `SwapQuoterOpts` type.

The two ways to fix this are:
- Remove the `: SwapQuoterOpts` casting as I described.  This forces TS to consider `DEFAULT_SWAP_QUOTER_OPTS` to be of the type of the object it is assigned -- which has `rfqt` populated
- Change `DEFAULT_SWAP_QUOTER_OPTS` type assignment to not be `SwapQuoterOpts` but instead be something like `SwapQuoterOptsWithRequiredRfqtParam` -- but there's not a quick utility type to do this

Reply:
FWIW I'm ok with this as-is if need be -- this shouldn't be a blocker to getting hte PR merged

Reply:
I still feel weird about removing that type annotation on the constant declaration.  I think I found a no-so-verbose way to restore type safety to this: https://github.com/0xProject/0x-monorepo/pull/2574/commits/352b1b43f2523a307689d91755734533679c31f8

Please let me know what you think